tests/rkt_notify_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build host coreos src kvm
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	sd_dbus "github.com/coreos/go-systemd/dbus"
+	sd_util "github.com/coreos/go-systemd/util"
+	"github.com/coreos/rkt/tests/testutils"
+	"github.com/godbus/dbus"
+)
+
+func TestNotify(t *testing.T) {
+	if !sd_util.IsRunningSystemd() {
+		t.Skip("Systemd is not running on the host.")
+	}
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	conn, err := sd_dbus.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	imageFile := getInspectImagePath()
+
+	image, err := filepath.Abs(imageFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// we need to add --silent-sigterm so inspect terminates correctly and the
+	// transient service exits successfully so it disappears from the list of
+	// units
+	sleepTime := 3
+	opts := fmt.Sprintf("-- --silent-sigterm --notify --sleep=%d", sleepTime)
```

Comment:
At line 94, the test is waiting some time because the service will be market as `activating` for some not  deterministic time. If I add `--pre-sleep` the `READY=1` notification will happen a `not deterministic number of seconds + pre-sleep-seconds`.

`--sleep` is there to be sure that the test catches the `active` status before the service terminates.

The fact the the sleep at line 94 and the sleep done by `inspect` are the same value is totally unrelated, I had that variable I reused it.